neuralop/losses/meta_losses.py

```diff
@@ -62,28 +65,123 @@ def __call__(self, pred: torch.Tensor, truth: torch.Tensor, **kwargs):
         else:
             return loss
 
-
 class WeightedSumLoss(object):
     """
     Computes an average or weighted sum of given losses.
     """
 
-    def __init__(self, losses, weights=None):
+    def __init__(self, losses, weights=None, return_sum=False):
         super().__init__()
         if weights is None:
             weights = [1.0 / len(losses)] * len(losses)
         if not len(weights) == len(losses):
             raise ValueError("Each loss must have a weight.")
-        self.losses = list(zip(losses, weights))
+        self.losses = {x.__class__.__name__: [x,y] for x,y in zip(losses,weights)}
+        self.return_sum = return_sum
 
     def __call__(self, *args, **kwargs):
-        weighted_loss = 0.0
-        for loss, weight in self.losses:
-            weighted_loss += weight * loss(*args, **kwargs)
-        return weighted_loss
+
+        loss_dict = {}
+        weight_dict = {}
+        for name, (loss,weight) in self.losses.items():
+            loss_value = loss(*args, **kwargs)
+            if self.return_sum:
+                weighted_loss += weight * loss_value
+            else:
+                loss_dict[name] = loss_value
+                weight_dict[name] = weight
+        if self.return_sum:
+            return weighted_loss
+        else:
+            return SumLossOutput(loss_dict, weight_dict)
 
     def __str__(self):
         description = "Combined loss: "
-        for loss, weight in self.losses:
-            description += f"{loss} (weight: {weight}) "
+        for name, (loss, weight) in self.losses.items():
+            description += f"{name} (weight: {weight}) "
         return description
+    
+class SumLossOutput(dict):
```

Comment:
What is the benefit of introducing `SumLossOutput`? 
Could we also just let every loss output the loss as well as additional `metrics`, which could be an empty dict but also, e.g., include individual losses, additional training metrics, etc.?